docs: begin official word list for Keptn documentation (#2049)
keptn/lifecycle-toolkit: docs/content/en/contribute/docs/word-list/_index.md

The Keptn project does not maintain a formal style guide
but should adhere to recommendations in:

* [CNCF Style Guide](https://github.com/cncf/foundation/blob/main/style-guide.md).

* The [Kubernetes documentation](https://kubernetes.io/docs/home/)
  is a good reference for Kubernetes terms.
   In particular:

  * [Concepts](https://kubernetes.io/docs/concepts/)
  * [Reference](https://kubernetes.io/docs/reference/)

* The [Google developer documentation style guide](https://developers.google.com/style)
  is a comprehensive reference for technical writers.
   In particular:

  * [Word list](https://developers.google.com/style/word-list)
    includes good information about words and phrases
    that are commonly used in technical documentation

## Keptn project

This is the proper name of the project that was developed
under the code name of "Keptn Lifecycle Toolkit (KLT)".
The earlier project is called "Keptn v1".

* Keptn is capitalized when used in prose as the name of the project,
  although the logo uses a lowercase "k".
  Use `keptn` if it is part of a command name, pathname,
  an argument to a command or function, etc.

> * As a project name that may be trademarked,

Reviewer: ```suggestion
* As a project name that is trademarked,
```

  you should not use an apostrophe-s to make it a possessive ("Keptn's")
  or hyphentate it (as in "Keptn-specific").

The Keptn project is a "toolkit" with three use cases, named:

* Metrics (or Deployment data access)

* Observability (or Deployment observability)

* Release lifecycle management (or Orchestrate deployment checks)

## CRD, resource, etc

Keptn makes extensive use of Kubernetes

> [Custom resources](https://developers.google.com/style/word-list).

Reviewer 2: ```suggestion
[Custom resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
```

Author: already fixed

> [Custom resources](https://developers.google.com/style/word-list).

Reviewer: this looks like the wrong link...

Author: It is fixed

It is important to use the related terminology correctly:

* A "Resource Definition (RD)" is the definition (or syntax)

>   of a resource that is part of the Kubernetes core

Reviewer 2: ```suggestion
  of a resource that is part of the official Kubernetes API
```
